### PR TITLE
Create PULL_REQUEST_TEMPLATE.md

### PR DESCRIPTION
**Pull request by dalonsoa**
A template to ensure that all contributions to the code follow some consistency.

If approved and merged, this should be implemented in all other projects we have (copy/paste will do).

### CODE REVIEW

**Review by jamesturner246 — APPROVED** [review]
Looks good. Is there any way to enforce this -- i.e. block an incomplete report -- or would that just be annoying?

**Comment by dalonsoa** [comment]
I don't know if it can be enforced, but I don't think we can enforce it, either. This is a guide so we remember the stuff. And there might be perfectly legitimate reasons to skip some of these steps.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Optimization (non-breaking, back-end change that speeds up the code)
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] Breaking change (whatever its nature)
+
+## Key checklist
+
+- [ ] All tests pass
+- [ ] The documentation builds and looks OK
+- [ ] Pre-commit hooks have be run successfully
```
Comment (alexdewar) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] Pre-commit hooks run successfully
```

**Review by alexdewar — APPROVED** [review]
LGTM, except for one typo.

Not all of the things in the checklist are relevant for this repo (e.g. the documentation doesn't need building), but I don't think that matters much.

**Review by AdrianDAlessandro — APPROVED** [review]
Appears generic enough that it can be used in most repos

**Comment by dalonsoa** [comment]
I think this is OK precisely because it is general, but when copied to a specific repo, the relevant checks should be updated to reflect what the repo needs.

For example, for a repo of a Python tool, using Sphinx for the documentation and the `pre-commit` tool to manage the pre-commit hooks, this could look:

``` markdown
- [ ] All tests pass: `python -m pytest`
- [ ] The documentation builds and looks OK: `python -m sphinx -b html docs docs/build`
- [ ] Pre-commit hooks run successfully: `pre-commit run --all-files`
```

**Comment by alexdewar** [comment]
That makes sense. I wasn't thinking before :-/

**Review by TinyMarsh — APPROVED** [review]

**Comment by TinyMarsh** [comment]
Cool, looks good!

**Pull request merged**